pyro/distributions/zero_inflated_poisson.py

```diff
@@ -27,13 +27,13 @@ def log_prob(self, value):
         gate, rate, value = broadcast_all(self.gate, self.rate, value)
         log_prob = (-gate).log1p() + (rate.log() * value) - rate - (value + 1).lgamma()
         zeros = value == 0
-        log_prob[zeros] = (gate[zeros] + log_prob[zeros].exp()).log()
+        log_prob = torch.where(zeros, (gate + log_prob.exp()).log(), log_prob)
```

Comment:
This is the place where autograd was complaining about in place mutation of variable that requires grad.

Reply:
Your change looks more reasonable to me.